cmd/pulumi-test-language/tests.go

```diff
@@ -28,19 +28,24 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-type languageTest struct {
+type testRun struct {
+	subDir string
 	config config.Map
-	// TODO: This should be a function so we don't have to load all providers in memory all the time.
-	providers []plugin.Provider
 	// This can be used to set a main value for the test.
 	main string
 	// TODO: This should just return "string", if == "" then ok, else fail
 	assert func(*L, result.Result, *deploy.Snapshot, display.ResourceChanges)
-
 	// updateOptions can be used to set the update options for the engine.
 	updateOptions engine.UpdateOptions
 }
 
+type languageTest struct {
+	// TODO: This should be a function so we don't have to load all providers in memory all the time.
+	providers []plugin.Provider
+
+	runs []testRun
```

Comment:
Happy to take naming suggestions here, I struggled with naming here.

Reply:
runs seems fine to me